_projects/2024/sunpy/sunpy-soar.md

```diff
@@ -0,0 +1,64 @@
+---
+name: Querying SOAR Metdata
+desc: Adding proper metadata handling to sunpy-soar
+requirements:
+ - Experience with Python
+difficulty: medium to high
+issues:
+ - https://github.com/sunpy/sunpy-soar/issues/46
+ - https://github.com/sunpy/sunpy-soar/issues/66
+mentors:
+ - hayesla
+ - nabobalis
+initiatives:
+ - GSOC
+project_size:
+ - 175 h / medium
+tags:
+ - sunpy
+ - Python
+collaborating_projects:
+ - sunpy
+---
+
+# Description
+
+[sunpy-soar](https://github.com/sunpy/sunpy-soar) is an extension to sunpy [Fido downloader](https://docs.sunpy.org/en/stable/tutorial/acquiring_data/index.html#sunpy-tutorial-acquiring-data-index) to download data from the Solar Orbiter Archive
+(SOAR).
+
+This data source has an extensive API and rich metadata which currently is unsupported by sunpy-soar.
+
+The goal of this GSoC project, is to add support for the large range of available metadata.
+The API of of the SOAR uses a [TAP protocol](https://git.ias.u-psud.fr/spice/data-analysis-club/-/blob/main/20231205-soar-tap/tap-demo.ipynb)  so it will be important for any student to get familiar with this protocol.
+Further to this, there is a large range of [possible values and types of tables](https://www.cosmos.esa.int/web/soar/tables-views-and-columns) which are important to understand as they are different for different files one might want to download.
+
+This involves understanding the [complicated attribute system](https://docs.sunpy.org/en/stable/topic_guide/extending_fido.html) that Fido uses.
+We will need to add new ones to query new metadata fields that the SOAR uses but also to translate them from TAP queries back into "normal" values after.
```

Comment:
In line with [this comment of sunpy-soar #46](https://github.com/sunpy/sunpy-soar/issues/46#issuecomment-1553168193), as we are not trying the impossible task of connecting all SOAR metadata to Fido attributes, could we maybe be more specific on the number of existing Fido attributes to connect ([some ideas](https://github.com/sunpy/sunpy-soar/issues/46#issuecomment-1557936562)), and perhaps also which additional attributes we will need? (at first sight I am not sure that we need to have many more than the existing Fido attributes: users can search with the existing attributes, and then filter the results list)

Reply:
Fair point, I have revised this. 

See if you prefer it now.

Reply:
Good for me in the new version, thanks!